server/src/main/java/keywhiz/service/daos/SecretDAO.java

```diff
@@ -171,17 +168,19 @@ public long createOrUpdateSecret(String name, String encryptedSecret, String cre
     return Optional.empty();
   }
 
-  /** @return all existing secrets. */
-  public ImmutableList<SecretSeriesAndContent> getSecrets() {
+  /** @return list of secrets. can limit/sort by expiry, and for group if given */
+  public ImmutableList<SecretSeriesAndContent> getSecrets(boolean sortByExpiry, Long expireMaxTime, Group group) {
```

Comment:
Make this `@Nullable Long expireMaxTime` (or if it's not nullable, make it `long` instead of `Long`).